### ISSUE TITLE
Error running `metadata` on Mac under Mono

### ISSUE THREAD

**Issue by joelmartinez** [dotnet/docfx#1720]
### Functional impact
I am unable to generate API documentation

### Minimal repro steps
using [version 2.18.2](https://github.com/dotnet/docfx/releases/tag/v2.18.2) of docfx, I run the following command on macOS:
```bash
$ mono ~/Downloads/docfx/docfx.exe metadata ~/dev/xamarin/api-doc-tools/apidoctools.sln 
```
### Expected result
I expect that API metadata would be generated

### Actual result
The resulting error is:
```
Could not load signature of Microsoft.CodeAnalysis.SymbolSearch.SymbolSearchUpdateEngine+RemoteControlService:CreateClient due to: Could not load file or assembly 'Microsoft.VisualStudio.RemoteControl, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. assembly:Microsoft.VisualStudio.RemoteControl, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a type:<unknown type> member:<none>
Could not load signature of Microsoft.CodeAnalysis.SymbolSearch.IRemoteControlService:CreateClient due to: Could not load file or assembly 'Microsoft.VisualStudio.RemoteControl, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. assembly:Microsoft.VisualStudio.RemoteControl, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a type:<unknown type> member:<none>
[17-06-01 07:42:47.346]Warning:Error opening solution /Users/jmartinez/dev/xamarin/api-doc-tools/apidoctools.sln: Could not resolve the signature of a virtual method. Ignored.
[17-06-01 07:42:47.377]Warning:No metadata is generated for .
[17-06-01 07:42:47.539]Info:Completed executing in 841.6077 milliseconds.


Build succeeded with warning.
[17-06-01 07:42:47.543]Warning:Error opening solution /Users/jmartinez/dev/xamarin/api-doc-tools/apidoctools.sln: Could not resolve the signature of a virtual method. Ignored.
[17-06-01 07:42:47.543]Warning:No metadata is generated for .
	2 Warning(s)
	0 Error(s)
```

### Further technical details
MacOS Sierra: v10.12.5

```bash
$ mono --version
Mono JIT compiler version 5.2.0.104 (2017-04/4a0006f Wed May 10 05:06:51 EDT 2017)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          yes(3.6.0svn-mono-master/8b1520c)
	GC:            sgen (concurrent by default)
```

**Comment by joelmartinez** [comment]
incidentally ... I also could not run this application using dotnet core

```bash
$ dotnet ~/Downloads/docfx/docfx.exe metadata ~/dev/xamarin/api-doc-tools/apidoctools.sln 
A fatal error was encountered. The library 'libhostpolicy.dylib' required to execute the application was not found in '/Users/jmartinez/Downloads/docfx'.
```

**Comment by vicancy** [comment]
Hi @joelmartinez  did the older version work for your solution?
Is it that your specific sln not working or a simple project from `docfx init -q` also not working? If the former, could you provide your sln for us to take a look?

**Comment by vicancy** [comment]
#1708

**Comment by vicancy** [comment]
https://github.com/dotnet/docfx/issues/1559#issuecomment-294735921 shows 2.16 works with some workarounds, however 2.17+ is broken again with a different error message.

**Comment by vicancy** [comment]
The error is fixed in v2.18.3

**Comment by joelmartinez** [comment]
Excellent news ... thanks team! 

/cc @migueldeicaza ^^
